kitsune/users/auth.py

```diff
@@ -195,30 +197,36 @@ def update_user(self, user, claims):
         # Check if the user has active subscriptions
         subscriptions = claims.get("subscriptions", [])
 
+        # The request attribute might not be set.
+        request = getattr(self, "request", None)
+
         if not profile.is_fxa_migrated:
             # Check if there is already a Firefox Account with this ID
             if Profile.objects.filter(fxa_uid=fxa_uid).exists():
-                msg = _("This Firefox Account is already used in another profile.")
-                messages.error(self.request, msg)
+                if request:
+                    msg = _("This Firefox Account is already used in another profile.")
+                    messages.error(request, msg)
                 return None
 
             # If it's not migrated, we can assume that there isn't an FxA id too
             profile.is_fxa_migrated = True
             profile.fxa_uid = fxa_uid
-            # This is the first time an existing user is using FxA. Redirect to profile edit
-            # in case the user wants to update any settings.
-            self.request.session["oidc_login_next"] = reverse("users.edit_my_profile")
-            messages.info(self.request, "fxa_notification_updated")
+            if request:
+                # This is the first time an existing user is using FxA. Redirect to profile edit
+                # in case the user wants to update any settings.
+                request.session["oidc_login_next"] = reverse("users.edit_my_profile")
+                messages.info(request, "fxa_notification_updated")
 
         # There is a change in the email in Firefox Accounts. Let's update user's email
         # unless we have a superuser
         if user.email != email and not user.is_staff:
             if User.objects.exclude(id=user.id).filter(email=email).exists():
```

Comment:
```suggestion
            if request and User.objects.exclude(id=user.id).filter(email=email).exists():
```

Reply:
I decided against making this change because we still want to check for an existing user having that email even if request is `None`.